README.md

```diff
@@ -330,753 +277,454 @@ Rails.application.routes.draw do
 
-But that link won't work just yet!
+We want to support a PATCH request to `/tasks/:id` where the user sends the fields to update in the request body.
 
-### Add a Route and Controller Action for an Individual Task
+### Creating the Route and Controller Action 
 
-In our `config/routes.rb` file, add the following route:
+In our `config/routes.rb` file, add the following route:
 
 **config/routes.rb**
 
 ```ruby
-Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
-  # Defines the root path route ("/")
-  # root "articles#index"
-  get "/", to: "welcome#index"
-
-  get "/tasks", to: "tasks#index"
-  get "/tasks/new", to: "tasks#new"
-  post "/tasks", to: "tasks#create"
-  get "/tasks/:id", to: "tasks#show"
-end
-```
-
-And in our `tasks_controller.rb` add a show action:
-
-**app/controllers/tasks_controller.rb**
-
-```ruby
-def show
-
-end
-```
-
-### Add a Show View
-
-Before we get to what will live inside this show action, let's dream up what we might want to be in our view for this action.
-
-In our views folder, create a file for `app/views/tasks/show.html.erb` and put the following HTML inside it:
-
-**app/views/tasks/show.html.erb**
-
-```html
-<a href="/tasks">Task Index</a>
-
-<h1><%= @task.title %></h1>
-
-<p><%= @task.description %></p>
+patch '/tasks/:id', to: 'tasks#update'
 ```
 
-Now, we have a show page that will show the information for a specific task, as well as a link back to the task index page.
-
-Let's see if we have accomplished our goal - navigate to [http://localhost:3000/tasks/1](http://localhost:3000/tasks/1) and let's see what happens.
-
-More errors! We should be seeing an error message telling us that we have a `NoMethodError in Tasks#show`, and if we look more closely, we are specifically seeing an `undefined method 'title' for nil:NilClass`. Thinking back to the errors we have seen in Mod1 - this is telling us that we are calling a method, `title` on a `nil`; in other words, `@task` is not holding on to a specific task that we can call `title` and `description` on.
-
-### Finding Specific Tasks
-
-Let's look back at our show action in our tasks controller and see if we can set up our instance variable `@task`. Remember when we talked about how we are inheriting some functionality from ActiveRecord that helps us interact with our database? Well, now's a great place to take advantage of the ActiveRecord method `find`, which will retrieve a record from our database based on that record's id. In this case, we can use find like this:
+And in our tasks controller, add the following action:
 
 **app/controllers/tasks_controller.rb**
 
+<!-- TODO, this does not work well if you only include title or description it wipes the other out. Not a true patch. -->
```

Comment:
We could do a `PUT` instead of a `PATCH` perhaps, or am I not fully thinking through the concern here?

Reply:
oooh this is tricky unless we use strong params, which we won't have taught them yet. Regardless of the method, I think this won't work as long as we're potentially passing nil values when the param key isn't present. 

I think we might need to do some conditional checks
new_title = params[:title] ? params[:title] : task.title

Reply:
This is ugly though.... need to think about this one

Reply:
Adding strong params to resolve this issue and start students off with best practices